### ISSUE TITLE
CompositeNavigationProp does not support navigation with the 'screen: screenName' route param

### ISSUE THREAD

**Issue by SocialBrothersDEV** [react-navigation/react-navigation#9017]
**Current Behavior**

- I am navigating to a screen in a nested navigator and getting typescript errors. 

**Expected Behavior**

- I expect to be able to specify params to my current route in CompositeNavigationProp and still have type checking.
- Or i expect to have a Typescript utility that allows me to use the `{screen: screenName}` for nested navigators param without specifying it everywhere manually.

**How to reproduce**

- Create a rootStack with a param list
- On one of the routes, create a nested navigator also with a param list (HomeStack for example).
- Make a type using the `StackNavigationProps<RootStackParamList, currentRoute>` type for the RootStack.
- Make a type using the `CompositeNavigationProp<StackNavigationProp<HomeStackParamList, 'Home'>, RootStackNavigationProp>` type for the HomeStack.
- Navigate to a screen within the HomeStack like this: 
```
navigation.navigate('Home', {
  screen: 'NewsOverview'
});
```
- You will get a typescript error  `Type '{ screen: string; }' is not assignable to type 'undefined'.`
- Add the screen parameters to the 'Home' key of the RootStackParamList like this: 
```
type RootStackParamList = {
  Home: {
    screen: keyof HomeStackParamList
  }
  Settings: undefined;
}
```
- In this case, you have lost type checking when the current route RootStackParamList will be 'Home'.
Eg: 

```
type HomeStackParamList = {
  NewsOverview: undefined;;
  NewsDetails: {
    ArticleId: string;
  }
} 

type RootStackParamList = {
  Home: {
    screen: keyof HomeStackParamList
  }
  Settings: undefined;
}

type HomeOverviewNavigationProp = CompositeNavigationProp<
  StackNavigationProp<HomeStackParamList, 'HomeOverview'>,
  StackNavigationProp<RootStackParamList, 'Home'> // Specifying a route here which has params in RootStackParamlist will cause the bug, because under the 'Home' key, you have navigation parameters (screen in this case).
>;

const navigateSomewhere = (navigation: HomeOverviewNavigationProp) => {
  navigation.navigate('Home', {
    screen: 'HomeOverview'
  }; // No autofill, no typechecks.
}
```
**Your Environment**

| software                       | version |
| ------------------------------ | ------- |
| iOS or Android                 | both
| @react-navigation/native       | 5.8.2
| react-native                   | 0.63.3
| expo                           | 
| node                           | 14.0
| npm or yarn                    | yarn

**Comment by satya164** [comment]
Works fine for me

![ezgif-6-33b6b1665705](https://user-images.githubusercontent.com/1174278/97908155-9cc4ee00-1d46-11eb-9a15-0386635ec2b2.gif)

**Comment by SocialBrothersDEV** [comment]
@satya164 Thanks for your reply. I did some digging after your findings, and concluded that the bug is not related to the structure of the screen parameters. It is related to the facts that the key for my stack was the same as the key for my screen in that stack. That is what broke the type checks. I have added a screen capture to clarify it.

However, is this the best way of supporting type checks for nested navigators? Can't this be generated automatically by the CompositeNavigationProp, or is the best way to actually specify the 'screen' parameter for every stack? 


![ezgif-6-a49eb4d2be20](https://user-images.githubusercontent.com/32843059/98108850-efb0b980-1e9c-11eb-841a-3ddba0130700.gif)
